client/my-sites/plans-single-products/index.jsx

```diff
@@ -2,33 +2,103 @@
  * External dependencies
  */
 import React from 'react';
-import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { isEnabled } from 'config';
 import Product from './product';
+import { getCurrentUserCurrencyCode } from 'state/current-user/selectors';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-const PlansSingleProducts = ( { translate } ) => {
+const PlansSingleProducts = props => {
+	const {
+		billingTimeFrame,
+		currencyCode,
+		isPlaceholder,
+		onProductSelect,
+		productProperties,
+	} = props;
 	const displayBackup = isEnabled( 'plans/jetpack-backup' );
 	const displayScan = isEnabled( 'plans/jetpack-scan' );
 
-	if ( ! displayBackup && ! displayScan ) {
-		return null;
-	}
-
 	return (
 		<div className="plans-single-products">
-			{ displayScan && <Product heading={ translate( 'Jetpack Scan' ) } /> }
-			{ displayBackup && <Product heading={ translate( 'Jetpack Backup' ) } /> }
+			{ displayScan && (
+				<Product
+					billingTimeFrame={ billingTimeFrame }
+					currencyCode={ currencyCode }
+					isPlaceholder={ isPlaceholder }
+					onSelect={ onProductSelect }
+					{ ...productProperties.jetpackScan }
+				/>
+			) }
+			{ displayBackup && (
+				<Product
+					billingTimeFrame={ billingTimeFrame }
+					currencyCode={ currencyCode }
+					isPlaceholder={ isPlaceholder }
+					onSelect={ onProductSelect }
+					{ ...productProperties.jetpackBackup }
+				/>
+			) }
 		</div>
 	);
 };
 
-export default localize( PlansSingleProducts );
+export default connect(
+	state => {
+		return {
+			billingTimeFrame: 'per year',
+			currencyCode: getCurrentUserCurrencyCode( state ),
+			isPlaceholder: false,
+			productProperties: {
+				jetpackScan: {
```

Comment:
I think you make a great point here. As I understand it we kept the store out of this Pr simple just because we want to get closer and figure out all the different components that we needed to build. But as you pointed out we still need to thing about the store and make sure that we props align better with what we have in the store.

Reply:
I claim that it should be the other way around because with the current approach we'll have to refactor half of these components just to make it work with the existing store. Building these components without the existing store in mind is like shooting in the dark, or choosing fruits while blindfolded.

Reply:
Just to clarify, @tyxla, this and the previous PR are not meant to provide a data layer in any way. The shape of the object and hardcoded/default values are here just to make it possible to work on the UI part of the project.

I'm aware that we'll have to refactor quite a bit of the code so that the props match the data shape we'll come up with later on. But since I'm not yet fully aware of how this is going to look like when we plug in data store, I took this simplified approach just to get us going.

Reply:
That's why I disagree with this: we already have working products and plans with our store. So we can envision how things could look if they accommodate the existing store already. I don't buy the need for additional development cost that the UI-first approach gives us.